src/lunabot_bringup/lunabot_bringup/preflight_check.py

```
@@ -76,9 +83,42 @@ def _load_yaml(path: Path) -> dict[str, Any]:
+    for parent in (search_root, *search_root.parents):
+        candidate = parent / CONTRACT_RELATIVE_PATH
+        if candidate.exists():
+            return candidate
+    return None
+
+
+def _package_contract_path() -> Path | None:
+    """Return the installed package copy of the contract when available."""
+    try:
+        share_dir = Path(get_package_share_directory("lunabot_bringup"))
+    except Exception:  # pragma: no cover - depends on sourced ROS install
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)